tests/wpt/web-platform-tests/html/semantics/forms/the-select-element/selected-index.html

```diff
@@ -0,0 +1,60 @@
+<!doctype html>
+<meta charset=utf-8>
+<title>HTMLSelectElement selectedIndex</title>
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<div id=log></div>
+
+<select id=empty></select>
+
+<select id=default>
+  <option></option>
+  <option></option>
+  <option></option>
+  <option></option>
+  <option></option>
+</select>
+
+<select id=disabled>
+  <option disabled></option>
+  <option></option>
+</select>
+
+<select id=selected>
+  <option></option>
+  <option selected></option>
+</select>
+
+<script>
+test(function () {
+  var select = document.getElementById('empty');
+  assert_equals(select.selectedIndex, -1);
+}, "get empty");
+
+test(function () {
+  var select = document.getElementById('default');
+  assert_equals(select.selectedIndex, 0);
```

Comment:
The algorithm for selecting an option also runs when any of the child options asks for a reset.